include/picongpu/particles/Particles.tpp

```diff
@@ -94,39 +100,38 @@ namespace picongpu
                 decltype(std::declval<T_ExchangeMemCfg>().DIR_SCALING_FACTOR),
                 typename T_ExchangeMemCfg::REF_LOCAL_DOM_SIZE>>
         {
-            static floatD_64 get()
+            static floatD_64 getOrtho()
             {
                 auto baseLocalCells = T_ExchangeMemCfg::REF_LOCAL_DOM_SIZE::toRT();
-                auto userScalingFactor = T_ExchangeMemCfg{}.DIR_SCALING_FACTOR;
-
                 auto localDomSize = Environment<simDim>::get().SubGrid().getLocalDomain().size;
-                // set too local domain size in case there is no base volume defined
+
+                auto scale = floatD_64::create(1.0);
                 for(uint32_t d = 0; d < simDim; ++d)
                 {
+                    // set to local domain size in case there is no base volume defined
                     if(baseLocalCells[d] <= 0)
                         baseLocalCells[d] = localDomSize[d];
+
+                    scale[d] = std::max(float_64(localDomSize[d]) / float_64(baseLocalCells[d]), 1.0);
                 }
 
-                auto scale = floatD_64::create(1.0);
+                return scale;
+            }
+
+            static floatD_64 getPara()
+            {
+                auto userScalingFactor = T_ExchangeMemCfg{}.DIR_SCALING_FACTOR;
+                floatD_64 scale;
                 for(uint32_t d = 0; d < simDim; ++d)
-                {
-                    auto dir1 = (d + 1) % simDim;
-                    auto dir2 = (d + 2) % simDim;
-                    // precision: numbers are small, therefore the usage of double is fine
-                    auto scaleDirection = std::ceil(
-                        float_64(localDomSize[dir1]) / float_64(baseLocalCells[dir1]) * float_64(localDomSize[dir2])
-                        / float_64(baseLocalCells[dir2]));
-                    float_64 scalingFactor = scaleDirection * userScalingFactor[d];
-                    // do not scale down
-                    scale[d] = std::max(scalingFactor, 1.0);
-                }
+                    scale[d] = userScalingFactor[d];
 
                 return scale;
```

Comment:
`pmacc::math::Vector` desperately needs a `Vector(const std::array<T_Type, T_dim>& )`, or does a utility for this purpose exist somewhere?

Reply:
Then probably not from an `std::array`, but a template constructor from any indexable type? Would also cover std::vector, etc.

Reply:
`std::vector` does not have a compile-time size, that c'tor would need to work differently. A c'tor from `std::array` could also be `constexpr`, btw.